pkg/controller/direct/asset/savedquery_controller.go

```diff
@@ -0,0 +1,375 @@
+	return nil, nil
+}
+
+type savedQueryAdapter struct {
+	gcpClient *gcp.Client
+	id        *krm.SavedQueryIdentity
+	desired   *krm.AssetSavedQuery
+	actual    *pb.SavedQuery
+	reader    client.Reader
+}
+
+var _ directbase.Adapter = &savedQueryAdapter{}
+
+func (a *savedQueryAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting asset saved query", "name", a.id)
+
+	name := a.id.String()
+	if a.actual != nil {
+		name = a.actual.Name
+	}
+	req := &pb.GetSavedQueryRequest{Name: name}
+	actual, err := a.gcpClient.GetSavedQuery(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting asset saved query %q from gcp: %w", a.id.String(), err)
+	}
+
+	a.actual = actual
+	return true, nil
+}
+
+func (a *savedQueryAdapter) normalizeReferences(ctx context.Context) error {
+	// No references to resolve in SavedQuery spec currently.
+	return nil
+}
+
+func (a *savedQueryAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating asset saved query", "name", a.id)
+	mapCtx := &direct.MapContext{}
+
+	if err := a.normalizeReferences(ctx); err != nil {
+		return fmt.Errorf("normalizing references: %w", err)
+	}
+
+	desired := a.desired.DeepCopy()
+	resource := AssetSavedQuerySpec_ToProto(mapCtx, &desired.Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	// resource.Name = a.id.String() // Name is output only for SavedQuery
+
+	req := &pb.CreateSavedQueryRequest{
+		Parent:       a.id.Parent().String(),
+		SavedQueryId: a.id.ID(),
+		SavedQuery:   resource,
+	}
+
+	// Attempt to create the saved query
+	actual, err := a.gcpClient.CreateSavedQuery(ctx, req)
+
+	// Regardless of CreateSavedQuery result (success or AlreadyExists),
+	// we need to ensure a.actual is populated by calling Find.
+	found := false
+	var findErr error
+
+	if err == nil {
+		log.V(2).Info("successfully created asset saved query in gcp, attempting to fetch it", "name", a.id)
+		a.actual = actual
+		found = true
+	} else if errors.IsAlreadyExists(err) {
+		// Resource already exists. Log and attempt to Find to populate a.actual.
+		log.V(2).Info("asset saved query already exists during creation attempt, attempting to fetch it", "name", a.id.String(), "warning", err)
+		found, findErr = a.Find(ctx)
+		if findErr != nil {
+			return fmt.Errorf("fetching existing asset saved query %q after CreateSavedQuery returned AlreadyExists failed: %w", a.id.String(), findErr)
+		}
+		if !found {
+			// This is unexpected if CreateSavedQuery returned AlreadyExists
+			return fmt.Errorf("asset saved query %q not found even though CreateSavedQuery returned AlreadyExists", a.id.String())
+		}
+	} else {
+		// Other error during creation
+		return fmt.Errorf("creating asset saved query %s: %w", a.id.String(), err)
+	}
+
+	// Update status
+	status := AssetSavedQueryStatus_FromProto(mapCtx, a.actual)
```

Comment:
We probably only need to use `AssetSavedQueryObservedState_FromProto` and set the `status.ExternalRef`, similar to how other controllers handle it. But this is not a blocker.